src/NumericalAlgorithms/Spectral/SwshDerivatives.hpp

```diff
@@ -0,0 +1,309 @@
+                                    typename DerivativeTag::derived_from>>(
+            pre_derivative_mode_tuple));
+  }
+};
+
+// Helper function for dealing with the parameter packs in the utilities which
+// evaluate several spin-weighted derivatives at once. The `apply` function of
+// this struct locates the like-spin set of paired modes and nodes, and calls
+// the member function of `SwshTransform` or `InverseSwshTransform` to perform
+// the spin-weighted transform. This function is a friend of both
+// `SwshTransform` and `InverseSwshTransform` to gain access to the private
+// `apply_to_vectors`.
+template <typename Transform, typename FullTagList>
+struct dispatch_to_transform;
+
+template <ComplexRepresentation Representation, typename... TransformTags,
+          typename... FullTags>
+struct dispatch_to_transform<
+    SwshTransform<tmpl::list<TransformTags...>, Representation>,
+    tmpl::list<FullTags...>> {
+  template <typename... ModalTypes, typename... NodalTypes>
+  static void apply(const size_t l_max, const size_t number_of_radial_points,
+                    const gsl::not_null<std::tuple<ModalTypes...>*> modal_tuple,
+                    const std::tuple<NodalTypes...>& nodal_tuple) noexcept {
+    SwshTransform<tmpl::list<TransformTags...>, Representation>::
+        apply_to_vectors(
+            get<tmpl::position_of_first<tmpl::list<FullTags...>,
+                                        TransformTags>>(*modal_tuple)...,
+            get<tmpl::position_of_first<tmpl::list<FullTags...>,
+                                        TransformTags>>(nodal_tuple)...,
+            l_max, number_of_radial_points);
+  }
+};
+
+template <ComplexRepresentation Representation, typename... TransformTags,
+          typename... FullTags>
+struct dispatch_to_transform<
+    InverseSwshTransform<tmpl::list<TransformTags...>, Representation>,
+    tmpl::list<FullTags...>> {
+  template <typename... NodalTypes, typename... ModalTypes>
+  static void apply(const size_t l_max, const size_t number_of_radial_points,
+                    const gsl::not_null<std::tuple<NodalTypes...>*> nodal_tuple,
+                    const std::tuple<ModalTypes...>& modal_tuple) noexcept {
+    InverseSwshTransform<tmpl::list<TransformTags...>, Representation>::
+        apply_to_vectors(
+            get<tmpl::position_of_first<tmpl::list<FullTags...>,
+                                        TransformTags>>(*nodal_tuple)...,
+            *get<tmpl::position_of_first<tmpl::list<FullTags...>,
+                                         TransformTags>>(modal_tuple)...,
+            l_max, number_of_radial_points);
+  }
+};
+
+// template 'implementation' for the DataBox mutate-compatible interface to
+// spin-weighted derivative evaluation. This impl version is needed to have easy
+// access to the `DeDuplicatedDerivedFromTagList` as a parameter pack
+template <typename DerivativeTagList, typename DeDuplicatedDerivedFromTagList,
```

Comment:
use something with 'differentiate' rather than 'derive' to avoid confusion with inheritance.